sdk/keyvault/azure-security-keyvault/azure/security/keyvault/certificates/_client.py

```diff
@@ -0,0 +1,222 @@
+        pass  # TODO
+
+    def __init__(self, vault_url, credentials, config=None, api_version=None, **kwargs):
+        if not credentials:
+            raise ValueError("credentials")
+
+        if not vault_url:
+            raise ValueError("vault_url")
+
+        self._vault_url = vault_url
+
+        if api_version is None:
+            api_version = KeyVaultClient.DEFAULT_API_VERSION
+
+        config = config or KeyVaultClient.get_configuration_class(api_version)(credentials)
+
+        # TODO generated default pipeline should be fine when token policy isn't necessary
+        policies = [
+            config.headers_policy,
+            config.user_agent_policy,
+            config.proxy_policy,
+            _BearerTokenCredentialPolicy(credentials),
+            config.redirect_policy,
+            config.retry_policy,
+            config.logging_policy,
+        ]
+        transport = RequestsTransport(config)
+        pipeline = Pipeline(transport, policies=policies)
+
+        self._client = KeyVaultClient(credentials, api_version=api_version, pipeline=pipeline)
+
+    @property
+    def vault_url(self):
+        # type: () -> str
+        return self._vault_url
+
+    def create_certificate(self, name, policy=None, enabled=None, not_before=None, expires=None, tags=None, **kwargs):
+        # type: (str, Optional[CertificatePolicy], Optional[bool], Optional[datetime], Optional[datetime], Optional[Dict[str, str]], Mapping[str, Mapping[str, Any]]) -> CertificateOperation
+        pass
+
+    def get_certificate(self, name, version=None, **kwargs):
+        # type: (str, Optional[str]) -> Certificate
+        pass
+
+    def delete_certificate(self, name, **kwargs):
+        # type: (str) -> DeletedCertificate
+        pass
+
+    def get_deleted_certificate(self, name, **kwargs):
+        # type: (str) -> DeletedCertificate
+        pass
+
+    def purge_deleted_certificate(self, name, **kwargs):
+        # type: (str) -> None
+        pass
+
+    def recover_deleted_certificate(self, name, **kwargs):
+        # type: (str) -> Certificate
+        pass
+
+    def import_certificate(
+        self,
+        name,
+        base64_encoded_certificate,
+        password=None,
+        policy=None,
+        enabled=None,
+        not_before=None,
+        expires=None,
+        tags=None,
+        **kwargs
+    ):
+        # type: (str, str, Optional[str], Optional[Policy], Optional[bool],Optional[datetime], Optional[datetime], Optional[Dict[str, str]], Mapping[str, Any]) -> Certificate
+        pass
+
+    def get_certificate_policy(self, name, **kwargs):
+        # type: (str) -> CertificatePolicy
+        pass
+
+    def update_certificate_policy(self, name, policy, enabled=None, not_before=None, expires=None, tags=None, **kwargs):
+        # type: (str, CertificatePolicy) -> CertificatePolicy
+        pass
+
+    def update_certificate(self, name, version, not_before=None, expires=None, enabled=None, tags=None, **kwargs):
```

Comment:
Is `not_before` a `boolean`? Or is the `#type` annotation out-of-date?

Reply:
updated.